go.mod

```diff
@@ -221,9 +221,18 @@ replace (
 
 // HV2 related packages
 replace (
-	cosmossdk.io/simapp => ./simapp
+	cosmossdk.io/client/v2 => github.com/0xPolygon/cosmos-sdk/client/v2 v2.0.0-beta.1.0.20240409091007-84d9b34a8681
+	cosmossdk.io/errors => github.com/0xPolygon/cosmos-sdk/errors v1.0.0-beta.7.0.20240409091007-84d9b34a8681
+	cosmossdk.io/simapp => github.com/0xPolygon/cosmos-sdk/simapp v0.0.0-20240409091007-84d9b34a8681
+	cosmossdk.io/store => github.com/0xPolygon/cosmos-sdk/store v1.0.2-0.20240409091007-84d9b34a8681
+	cosmossdk.io/x/circuit => github.com/0xPolygon/cosmos-sdk/x/circuit v0.1.1-0.20240409091007-84d9b34a8681
+	cosmossdk.io/x/evidence => github.com/0xPolygon/cosmos-sdk/x/evidence v0.1.1-0.20240409091007-84d9b34a8681
+	cosmossdk.io/x/feegrant => github.com/0xPolygon/cosmos-sdk/x/feegrant v0.1.1-0.20240409091007-84d9b34a8681
+	cosmossdk.io/x/nft => github.com/0xPolygon/cosmos-sdk/x/nft v0.1.1-0.20240409091007-84d9b34a8681
+	cosmossdk.io/x/tx => github.com/0xPolygon/cosmos-sdk/x/tx v0.8.1-0.20240409091007-84d9b34a8681
+	cosmossdk.io/x/upgrade => github.com/0xPolygon/cosmos-sdk/x/upgrade v0.1.2-0.20240409091007-84d9b34a8681
 	github.com/cometbft/cometbft => github.com/0xPolygon/cometbft v0.1.0-beta
-	github.com/cosmos/cosmos-sdk => github.com/0xPolygon/cosmos-sdk v0.50.3-0.20240409090657-2e910ae577a9
+	github.com/cosmos/cosmos-sdk => github.com/0xPolygon/cosmos-sdk v0.1.0-beta-polygon
```

Comment:
Everytime we have a new release, we will need to update this, right? Wondering if there's a different way this can be done.

Reply:
Yeah, not sure about alternatives

Reply:
Check my comment above about the relative path. It can be used during development. Then, when you release a version of the fork, you just need to remember to come here and set the official version

Reply:
Hmmm, actually, are you sure you need this line in particular. What breaks if you delete it?

Reply:
I'd need to double check this, but iirc, the following happens if I remove that `replace`.
While importing `cosmos-sdk` as a dependency in `heimdall-v2` I get `parsing go.mod module declares its path as "github.com/cosmos/cosmos-sdk" but was required as "github.com/0xPolygon/cosmos-sdk"`

Reply:
Anyway using the relative path now, but I confirm that was the issue
Addressed here https://github.com/0xPolygon/cosmos-sdk/pull/16/commits/1554abab5ff582a27ef21c3d61f33e8494f98801

Reply:
The relative path is good, I think, since it removes the hassle of having to be changing it all the time while in heavy development.

But, I'm still intrigued by the error you're getting. For context, what motivated my comment initially was the fact of seeing that the very first version of CometBFT (when we forked from Tendermint Core) declares itself as `github.com/tendermint/tendermint` see [here](https://github.com/cometbft/cometbft/blob/v0.34.x/go.mod#L1) , and so, there is no `require` of `tendermint/tendermint` anywhere in that `go.mod`.

I also checked [Celestia's fork](https://github.com/celestiaorg/celestia-core/blob/main/go.mod#L1), which is now a fork of CometBFT. Same situation.

So, since you declare your repo in the same way (`module github.com/cosmos/cosmos-sdk`), I thought you wouldn't need to include it in the `require` section.

Just to be clear, I'm only referring to the line this thread is attached to (the top level `go.mod`). For all other `go.mod` files, the relative path is the best approach.

Reply:
Thanks @sergio-mena 
Just created the release, will re-check this while importing in `heimdall-v2` and - in case - make the required change

Reply:
Resolved